CMakeLists.txt

```diff
@@ -55,10 +56,30 @@ endif()
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
-# For code, link against the static crt on windows
 if ("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    # Link against the static crt
     include(MSVCStaticCrt)
+
+    # Turn off incremental linking
     include(MSVCLinkerFlags)
+
+    # Enable source linking
+    # NOTE: Dependencies are not properly setup here.
+    #       Currently, CMake does not know to re-link if SOURCE_LINK_JSON changes
+    #       Currently, CMake does not re-generate SOURCE_LINK_JSON if git's HEAD changes
+    if (K4A_SOURCE_LINK)
+        if ("${CMAKE_C_COMPILER_VERSION}" VERSION_GREATER_EQUAL "19.20")
```

Comment:
Do we have plans around when we are going to switch to 2019? And when we'll no longer build on 2017?

Reply:
These are two different asks.

1. When are we going to release binaries built with VS2019 instead of binaries built with VS2017?
   This we can start doing soon. VS2017 and VS2019 have toolchains which (should) produce ABI compatible binaries. Therefore, we can release binaries built by VS2019 and VS2017 will still work.

2. When are we going to stop building VS2017?
    I imagine this will not be for awhile.